src/o365/graph/commands/subscriptions/subscription-add.ts

```diff
@@ -0,0 +1,244 @@
+import commands from '../../commands';
+import request from '../../../../request';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  resource: string;
+  changeType: string;
+  notificationUrl: string;
+  expirationDateTime: string;
```

Comment:
Properties that are optional, should be designated as such in the interface to make it clear in the code that they could be absent.